packages/coverage-v8/src/provider.ts

```diff
@@ -497,27 +486,49 @@ export class V8CoverageProvider extends BaseCoverageProvider implements Coverage
 
   private async convertCoverage(
     coverage: RawCoverage,
-    projectName?: ProjectName,
-    transformMode?: 'web' | 'ssr',
+    project: WorkspaceProject = this.ctx.getCoreWorkspaceProject(),
+    transformMode?: keyof CoverageFilesByTransformMode,
   ): Promise<CoverageMap> {
-    const viteNode
-      = this.ctx.projects.find(project => project.getName() === projectName)
-        ?.vitenode || this.ctx.vitenode
-    const fetchCache = transformMode
-      ? viteNode.fetchCaches[transformMode]
-      : viteNode.fetchCache
+    let fetchCache = project.vitenode.fetchCache
+
+    if (transformMode) {
+      fetchCache = transformMode === 'browser' ? new Map() : project.vitenode.fetchCaches[transformMode]
+    }
+
     const transformResults = normalizeTransformResults(fetchCache)
 
-    const scriptCoverages = coverage.result.filter(result =>
-      this.testExclude.shouldInstrument(fileURLToPath(result.url)),
-    )
+    async function onTransform(filepath: string) {
+      if (transformMode === 'browser' && project.browser) {
+        const result = await project.browser.vite.transformRequest(filepath.replace(project.config.root, ''))
+
+        if (result) {
+          return { ...result, code: `${result.code}// <inline-source-map>` }
+        }
+      }
+      return project.vitenode.transformRequest(filepath)
+    }
+
+    const scriptCoverages = []
+
+    for (const result of coverage.result) {
+      if (transformMode === 'browser') {
+        if (result.url.includes('@fs/')) {
+          result.url = `file://${result.url.replace('@fs/', '')}`
+        }
+        else {
+          result.url = `file://${project.config.root}${result.url}`
```

Comment:
`pathToFileURL(join(project.config.root, result.url))` - join doesn't behave like `resolve` and will keep all segments

Reply:
`pathToFleURL` escapes query parameters so it cannot be used here. 

```sh
$ node
> url.pathToFileURL("/fixtures/src/Vue/Counter/Counter.component.ts?vue&type=script&src=true&lang.ts")

URL {
  href: 'file:///fixtures/src/Vue/Counter/Counter.component.ts%3Fvue&type=script&src=true&lang.ts',
//                                                            ^^
  pathname: '/fixtures/src/Vue/Counter/Counter.component.ts%3Fvue&type=script&src=true&lang.ts',
  search: '',
}
```

And `join('file://', project.config.root, result.url)` doesn't work either as it converts to `file:/path` instead of `file:///path`